subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/DefaultExecutionStateChangeDetector.java

```diff
@@ -147,8 +149,8 @@ public IncrementalDetectedExecutionStateChanges(
         }
 
         @Override
-        public InputChangesInternal createInputChanges(ImmutableMultimap<Object, String> incrementalParameterNameByValue) {
-            return new IncrementalInputChanges(inputFileChanges, incrementalParameterNameByValue);
+        public InputChangesInternal createInputChanges(ImmutableMultimap<Object, String> incrementalParameterNamesByValue) {
```

Comment:
Does this still need to be a multimap? We said we'd fail on collision earlier.

Reply:
I'll change it.